samples/samples/admin/backup_snippet.py

```diff
@@ -0,0 +1,556 @@
+        parent=instance.name,
+        database_id=new_database_id,
+        backup="{}/backups/{}".format(instance.name, backup_id),
+        encryption_config=encryption_config,
+    )
+    operation = spanner_client.database_admin_api.restore_database(request)
+
+    # Wait for restore operation to complete.
+    db = operation.result(1600)
+
+    # Newly created database has restore information.
+    restore_info = db.restore_info
+    print(
+        "Database {} restored to {} from backup {} with using encryption key {}.".format(
+            restore_info.backup_info.source_database,
+            new_database_id,
+            restore_info.backup_info.backup,
+            restore_info.encryption_config.kms_key_name,
+        )
+    )
+
+
+# [END spanner_restore_backup_with_encryption_key]
+
+
+# [START spanner_cancel_backup_create]
+def cancel_backup(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    database = instance.database(database_id)
+
+    expire_time = datetime.utcnow() + timedelta(days=30)
+
+    # Create a backup.
+    request = backup_pb.CreateBackupRequest(
+        parent=instance.name,
+        backup_id=backup_id,
+        backup=backup_pb.Backup(
+            database=database.name,
+            expire_time=expire_time,
+        ),
+    )
+
+    operation = spanner_client.database_admin_api.create_backup(request)
+    # Cancel backup creation.
+    operation.cancel()
+
+    # Cancel operations are best effort so either it will complete or
+    # be cancelled.
+    while not operation.done():
+        time.sleep(300)  # 5 mins
+
+    try:
+        spanner_client.database_admin_api.get_backup(
+            backup_pb.GetBackupRequest(
+                name="{}/backups/{}".format(instance.name, backup_id)
+            )
+        )
+    except NotFound:
+        print("Backup creation was successfully cancelled.")
+        return
+    print("Backup was created before the cancel completed.")
+    spanner_client.database_admin_api.delete_backup(
+        backup_pb.DeleteBackupRequest(
+            name="{}/backups/{}".format(instance.name, backup_id)
+        )
+    )
+    print("Backup deleted.")
+
+
+# [END spanner_cancel_backup_create]
+
+
+# [START spanner_list_backup_operations]
+def list_backup_operations(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List the CreateBackup operations.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.CreateBackupMetadata) "
+        "AND (metadata.database:{})"
+    ).format(database_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
```

Comment:
Should we instead directly use code used here? https://github.com/googleapis/python-spanner/blob/3aab0ed5ed3cd078835812dae183a333fe1d3a20/google/cloud/spanner_v1/instance.py#L726
Our handwritten code returns the data as `google.api_core.operation.Operation` and customers could use this for other things like getting the operationid, deserialize etc.

In this example the `metadata` is specific field of  `google.api_core.operation.Operation`. Instead directly showing the conversion to `google.api_core.operation.Operation` will help customers,
WDYT?

Reply:
All the languages should be consistent in decoding operations for a specifc samples, we are trying to follow what is present in:
Golang= https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_list_backup_operations.go#L64

JAVA= https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/SpannerSample.java#L1683-L1689